drivers/sensor/icm42670S/icm42670S_aml.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2023 TDK Invensense
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/sensor/icm42670S.h>
+
+#include "icm42670S.h"
+#include "invn_algo_aml.h"
+
+LOG_MODULE_REGISTER(ICM42670S_AML, CONFIG_SENSOR_LOG_LEVEL);
+
+#ifdef CONFIG_ICM42670S_AML
+static int32_t accel_mounting_matrix[9]= {0,   -1,    0,
+                                         -1,    0,    0,
+                                          0,    0,    1 };
+
+static int32_t gyro_mounting_matrix[9]= { 0,    1,    0,
+                                          1,    0,    0,
+                                          0,    0,   -1 };
+
+
+int icm42670S_aml_init(const struct device *dev, inv_imu_device_t *s, int8_t delta_gain_x, int8_t delta_gain_y)
+{
+	int rc = 0;
+	/* Contains algorithms configuration */
+	InvnAlgoAMLConfig config;
+	struct icm42670S_data *data = dev->data;
+	
+	/* FSR */
+	config.acc_fsr = 16;
+	config.gyr_fsr = 2000;
+	data->accel_fs = 16;
+	data->gyro_fs = 2000;
+
+	/* Delta Gain */
+	config.delta_gain[0] = delta_gain_x;
+	config.delta_gain[1] = delta_gain_y;
+	
+	config.gestures_auto_reset = 1;
+
+	/* Initialize AML algorithms */
+	rc = invn_algo_aml_init(s, &config);
+	if (rc != 0) {
+		LOG_ERR("Error. Authentication Failed.");
+		return rc;
+	} else {
+		LOG_DBG("Algorithm init OK");
+	}
+	
+	/* Configure ICM device */
+	rc |= inv_imu_set_accel_fsr(s, ACCEL_CONFIG0_FS_SEL_16g);
+	rc |= inv_imu_set_gyro_fsr(s, GYRO_CONFIG0_FS_SEL_2000dps);
+
+	/* Set frequencies */ 
+	rc |= inv_imu_set_accel_frequency(s, ACCEL_CONFIG0_ODR_100_HZ);
+	rc |= inv_imu_set_gyro_frequency(s, GYRO_CONFIG0_ODR_100_HZ);
+
+	/* Enable sensors */
+	rc |= inv_imu_enable_accel_low_noise_mode(s);
+	rc |= inv_imu_enable_gyro_low_noise_mode(s);
+	
+	if (rc != 0) {
+		LOG_ERR("Error while configuring ICM device");
+	}
+	
+	return rc;
+}
+
+static void apply_mounting_matrix(const int32_t matrix[9], int16_t raw[3])
+{
+	unsigned i;
+	int16_t out_raw[3];
+	
+	for(i = 0; i < 3; i++) {
+		out_raw[i] =  ((int16_t)matrix[3*i+0] * raw[0]);
```

Comment:
If you want to simplify the cast, you might just set the mounting matrix as int16_t I guess